lib/api/objectPut.js

```diff
@@ -103,7 +103,13 @@ function objectPut(authInfo, request, streamingV4Params, log, callback) {
                     'putObject');
                 return callback(err, responseHeaders);
             }
-            const newByteLength = request.parsedContentLength;
+            const isIngest = (request.headers['x-amz-meta-mdonly']
+                && !isNaN(request.headers['x-amz-meta-size'])) || false;
```

Comment:
`Number.isNaN` is more predictible than the global `isNaN` btw as the global one does type conversion along with testing the value.